crates/holochain/src/core/workflow/call_zome_workflow/unit_tests.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{
+    conductor::api::CellConductorApi,
+    core::{
+        queue_consumer::TriggerSender,
+        ribosome::{real_ribosome::RealRibosome, ZomeCallInvocation},
+        workflow::{call_zome_workflow, CallZomeWorkflowArgs},
+    },
+    sweettest::{SweetConductor, SweetDnaFile},
+    test_utils::new_zome_call_params,
+};
+use ::fixt::fixt;
+use hdk::prelude::CellId;
+use holo_hash::fixt::ActionHashFixturator;
+use holochain_keystore::MetaLairClient;
+use holochain_p2p::{actor::MockHcP2p, HolochainP2pDna};
+use holochain_state::host_fn_workspace::SourceChainWorkspace;
+use holochain_wasm_test_utils::TestWasm;
+use kitsune2_api::DhtArc;
+use std::sync::Arc;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn trigger_integration_workflow_after_creating_ops() {
+    let mut hc_p2p = MockHcP2p::new();
+    hc_p2p
+        .expect_target_arcs()
+        .returning(|_| Box::pin(async { Ok(vec![DhtArc::FULL]) }));
+    let TestCase {
+        workspace,
+        network,
+        keystore,
+        args,
+        _conductor,
+    } = TestCase::new(hc_p2p, "create", ()).await;
+    let (trigger_publish_dht_ops, _) = TriggerSender::new();
+    let (trigger_integrate_dht_ops, mut integrate_dht_ops_rx) = TriggerSender::new();
+    let (trigger_countersigning, _) = TriggerSender::new();
+
+    let _ = call_zome_workflow(
+        workspace,
+        network,
+        keystore,
+        args,
+        trigger_publish_dht_ops,
+        trigger_integrate_dht_ops,
+        trigger_countersigning,
+    )
+    .await
+    .unwrap()
+    .unwrap();
+    // Assert the integration workflow has been triggered.
+    integrate_dht_ops_rx.listen().await.unwrap();
```

Comment:
Recommend timeout when asserting events that might not happen. Optional but makes the test easier to debug if it breaks